internal/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -270,14 +270,16 @@ func (r *Reconciler) reconcile(ctx context.Context, logger logr.Logger, cluster
 			Status:   corev1.ConditionFalse,
 			Severity: clusterv1.ConditionSeverityWarning,
 			Reason:   clusterv1.TooManyUnhealthyReason,
-			Message:  message,
+			// Format message to account for percent sign.
+			Message: strings.Replace(message, "%", "%%", -1),
 		})
 
 		r.recorder.Eventf(
 			m,
 			corev1.EventTypeWarning,
 			EventRemediationRestricted,
-			message,
+			// Format message to account for percent sign.
+			strings.Replace(message, "%", "%%", -1),
```

Comment:
If I got this correctly we need this because we're using Eventf which expects a messageFmt and args, should we just use `Event` instead?

Reply:
Much better solution 😄  Thanks for catching that.